packages/plugins/parcel-reporter-manifest/src/utils.ts

```diff
@@ -0,0 +1,105 @@
+import type { Asset, BundleGraph, PackagedBundle } from '@parcel/plugin';
+import type { Manifest } from '@react-loosely-lazy/manifest';
+import { relative } from 'path';
+
+const getAssetId = (basePath: string, assetPath: string) => {
+  let assetId = relative(basePath, assetPath);
+
+  if (assetId.charAt(0) !== '.') {
+    assetId = `./${assetId}`;
+  }
+
+  return assetId;
+};
+
+export type BuildManifestsOptions = {
+  bundleGraph: BundleGraph<PackagedBundle>;
+  options: {
+    assetBasePath: string;
+  };
+};
+
+export const buildManifests = ({
+  bundleGraph,
+  options: { assetBasePath },
+}: BuildManifestsOptions): Map<string, Manifest> => {
+  const manifestsByTarget = new Map<string, Manifest>();
+  const visitedAssetsByTarget = new Map<string, WeakSet<Asset>>();
+
+  for (const bundle of bundleGraph.getBundles()) {
+    const { distDir } = bundle.target;
+    if (!manifestsByTarget.has(distDir)) {
+      manifestsByTarget.set(distDir, {
+        publicPath: bundle.target.publicUrl,
+        assets: {},
+      });
+    }
+
+    if (!visitedAssetsByTarget.has(distDir)) {
+      visitedAssetsByTarget.set(distDir, new WeakSet<Asset>());
+    }
+
+    const manifest = manifestsByTarget.get(distDir)!;
+    const visitedAssets = visitedAssetsByTarget.get(distDir)!;
+
+    bundle.traverseAssets((asset, _, actions) => {
+      if (visitedAssets.has(asset)) {
```

Comment:
Is this still needed? I think traverseAssets should guarantee this. Was it leftover from something else?

Reply:
It was needed based on the discussions we had earlier, though I did have a stash of changes that use `bundleGraph.traverse` instead which avoids this altogether. But, I couldn't test it in Jira at the time since it wasn't on the latest parcel version.